In [1]:
import numpy as np

In [2]:
fh = open("michelson_data.txt", "r")
data = np.array([np.array(list(map(int,line.split()))) for line in fh.read().splitlines()])
avg_dat = np.array([np.append(data[i],np.average(data[i][1:])) for i in range(len(data[0:,]))])
fh.close()

### Apparatus
A laser of wavelength 535 nm has been used in the interferometer, and the thickness of the glass slab used for measuring the refractive index is 1 mm.

In [3]:
t = 1e-3
wv_l = 535e-9

### Calculation of Refractive Index

The refractive index can be caculated using the angle of incidence ($\theta$) and the fringe shifts ($m$) using the following formula :
$$\boxed{n_g = \cfrac{(1 - \cos\theta)\left(1 - \cfrac{m\lambda}{2t}\right)}{(1 - \cos{\theta}) - \cfrac{m\lambda}{2t}}}$$

Where $t$ is the thickness of the glass slab used, and $\lambda$ is the wavelength of the laser.

In [4]:
n_g = (1 - np.cos(avg_dat[:,11]*np.pi / 180))*(1 - ((avg_dat[:,0]*wv_l)/(2*t))) / ((1 - np.cos(avg_dat[:,11]*np.pi / 180)) - ((avg_dat[:,0]*wv_l)/(2*t)))
n_g

array([1.53531693, 1.56768698, 1.54630682, 1.50690853])

### Calculated refractive index of glass for diffirent fringe shifts


| Fringe Shifts| Avg. angle of incidence |Calculated refractive index|
|:---:|:---:|:---:|
| 20 | 10 | 1.54 |
| 30 | 12 | 1.57 |
| 40 | 14 | 1.55 |
| 50 | 16 | 1.51 |

In [5]:
print(np.average(n_g), np.std(n_g))

1.5390548143926375 0.02190746237649228


The value of the refractive index of glass thus obtained is $\boxed{1.53 \pm 0.02}$